src/volatile_memory.rs

```diff
@@ -146,6 +178,61 @@ pub trait VolatileMemory {
         }
     }
 
+    /// Gets a reference to T at `offset`.  The resulting pointer must be
+    /// aligned, or the function fails.
+    ///
+    /// To use this safely, the caller must guarantee that there are no other
+    /// users of the given chunk of memory for the lifetime of the result.
+    unsafe fn aligned_as_ref<T: ByteValued>(&self, offset: usize) -> Result<&T> {
```

Comment:
The body of these functions seems pretty boilerplate :-s Can we extract it into a helper method/macro and reuse that instead?

Reply:
I added check_alignment.